### PR TITLE
[Enhancement] Add extended status for FE managing BE/CN

### PR DESCRIPTION
**Pull request by kevincai**
## What I'm doing:

Refs #46652

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [x] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/system/BackendHbResponse.java [hunk 1]
```diff
@@ -63,6 +64,9 @@ public class BackendHbResponse extends HeartbeatResponse implements Writable {
     private int cpuCores;
     @SerializedName(value = "rebootTime")
     private long rebootTime = -1L;
+
+    @SerializedName(value = "statusCode")
+    private TStatusCode statusCode = TStatusCode.OK;
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/system/BackendHbResponse.java:69]: what is the content of TStatusCode type in serialization mode?

File: fe/fe-core/src/main/java/com/starrocks/system/ComputeNode.java [hunk 2]
```diff
@@ -116,6 +118,10 @@ public class ComputeNode implements IComputable, Writable {
     @SerializedName("isSetStoragePath")
     private volatile boolean isSetStoragePath = false;
 
+    // Tracking the heartbeat status, CONNECTING/ALIVE/SHUTDOWN/DISCONNECTED
+    @SerializedName("status")
+    private Status status;
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/system/ComputeNode.java:123]: Is the string literal of enum item serialized? That means the name can't be changed anymore. Better to add NOTE to the enum definition.

File: fe/fe-core/src/main/java/com/starrocks/system/BackendHbResponse.java (hunk 1 above)
Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/system/BackendHbResponse.java:69]: the new field will not be brought to the followers/observers giving current read/write implementation in BackendHbResponse.java. need to think about other ways to sync the status with followers/observers.

Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/system/BackendHbResponse.java:69]: in heartbeatV2, the heartbeat package is synced to the followers/observers with `HbPackage.toGson`, the statusCode can be correctly sync to the follower.

**Review by trueeyu — DISMISSED** [review]

**Review by starrocks-xupeng — APPROVED** [review]

**Comment by kevincai** [comment]
@mergifyio rebase

**Review by trueeyu — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Review by alvin-phoenix-ai — APPROVED** [review]

**Pull request merged**